pkg/cmd/server/start/start_master.go

```diff
@@ -429,7 +429,7 @@ func StartAPI(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) error {
 	}
 
 	// verify we can connect to etcd with the provided config
-	if err := etcd.TestEtcdClient(oc.EtcdClient); err != nil {
+	if _, err := etcd.GetAndTestEtcdClient(oc.Options.EtcdClientInfo); err != nil {
```

Comment:
does creating a new client leave an active keepalive connection cache in place? if so, not sure I want to construct a throwaway client here